libcudacxx/include/cuda/__memory_resource/cuda_memory_resource.h

```diff
@@ -32,15 +32,30 @@
 #  include <cuda/__memory_resource/resource.h>
 #  include <cuda/__memory_resource/resource_ref.h>
 #  include <cuda/std/__cuda/api_wrapper.h>
+#  include <cuda/std/__cuda/scoped_device.h>
 #  include <cuda/std/__new/bad_alloc.h>
 
 #  if _CCCL_STD_VER >= 2014
 
 _LIBCUDACXX_BEGIN_NAMESPACE_CUDA_MR
 
 //! @brief cuda_memory_resource uses `cudaMalloc` / `cudaFree` for allocation / deallocation.
-struct cuda_memory_resource
+//! By default uses device 0 to allocate memory
+class cuda_memory_resource
```

Comment:
important: This feels like an important design point that will have wide implications. 

As @wence-'s comment shows, always using device 0 may be astonishing to the average CUDA user who will typically expect when a device isn't specified that it will use the "current device". 

@pciolkosz @ericniebler what is the latest thinking with the notion of the "default device" in the runtime modernization work?

Reply:
This aligns with the current plan for the C++ runtime, where default device is always 0 and current device is ignored at all times

Reply:
FWIW, I would [love it](https://www.youtube.com/watch?v=mk87a7r0V60) if there were no such concept as a default device (I like that there is no "current device" concept), and that everything that cares about devices needs to explicitly be constructed on one.

Reply:
That's generally the direction we're headed in. With `cuda::mr` we are taking a more gradual approach since it's already in use and we don't want to abruptly break people by requiring a device at construction. The idea is that we're adding the default argument to the ctor now, and we'll then deprecate/remove it later.